scanapp/tests/test_serializer.py

```diff
@@ -0,0 +1,283 @@
+from django.utils import timezone
+
+from scanapp.models import Copyright
+from scanapp.models import CopyrightAuthor
+from scanapp.models import CopyrightHolder
+from scanapp.models import CopyrightStatement
+from scanapp.models import License
+from scanapp.models import Package
+from scanapp.models import Scan
+from scanapp.models import ScanError
+from scanapp.models import ScannedFile
+
+from scanapp.serializers import AllModelSerializerHelper
+from scanapp.serializers import AllModelSerializer
+
+
+class AllModelSerializerHelperTestCase(TestCase):
+    def test_all_model_serializer_helper(self):
+        user = User.objects.create_user(username='username', password='password')
+        scan = Scan.objects.create(
+            user=user,
+            url='https;//github.com',
+            scan_directory='media/username',
+            scancode_notice='Some dummy notice',
+            scancode_version='2.0.0rc3',
+            files_count=200,
+            scan_start_time=timezone.now(),
+            scan_end_time=timezone.now()
+        )
+        scanned_file = ScannedFile.objects.create(scan=scan, path='/home/nexb/server/')
+        license = License.objects.create(
+            scanned_file=scanned_file,
+            key='A',
+            score=78,
+            short_name='mit',
+            category='some category',
+            owner='mit',
+            homepage_url='https://github.com/',
+            text_url='http://github.com/',
+            dejacode_url='https://github.com',
+            spdx_license_key='mit',
+            spdx_url='https://github.com/',
+            start_line=21,
+            end_line=567,
+            matched_rule=json.loads('{"url": ["https://github.com", "https://google.com"]}')
+        )
+        copyright = Copyright.objects.create(
+            scanned_file=scanned_file,
+            start_line=800,
+            end_line=1000
+        )
+        copyright_holder = CopyrightHolder.objects.create(
+            copyright=copyright,
+            holder='Ranvir Singh'
+        )
+        copyright_statement = CopyrightStatement.objects.create(
+            copyright=copyright,
+            statement='mit copyright statement'
+        )
+        copyright_author = CopyrightAuthor.objects.create(
+            copyright=copyright,
+            author='Ranvir Singh'
+        )
+        package = Package.objects.create(scanned_file=scanned_file, package='bootstrap')
+        scan_error = ScanError.objects.create(
+            scanned_file=scanned_file,
+            scan_error='Integration Issue'
+        )
+
+        scan_serializer_helper = AllModelSerializerHelper(scan)
+        scan_id = scan.pk
+        self.assertEqual(scan_serializer_helper.scan, Scan.objects.get(pk=scan_id))
+        self.assertEqual(
+            str(scan_serializer_helper.scanned_file),
+            str(ScannedFile.objects.filter(scan=Scan.objects.get(pk=scan_id))))
+        self.assertEqual(
+            str(scan_serializer_helper.license),
```

Comment:
IMHO you should simplify your tests: store an expected JSON file with the nested serialized results....
Then  do you serialization and json.load the expected JSON to check it matches your serialization.
This would make the test code simpler and the results easier to visualize